.github/workflows/build.yml

```diff
@@ -13,3 +13,24 @@ jobs:
       target-name: 'dd_pprof' # target name in binding.gyp
       package-manager: 'npm' # npm or yarn
       cache: true # enable caching of dependencies based on lockfile
+
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    environment: npm
+    env:
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
+      - uses: actions/setup-node@v3
+        with:
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm install
+      - id: pkg
+        run: |
+          content=`cat ./package.json | tr '\n' ' '`
+          echo "::set-output name=json::$content"
+      - run: npm version --no-git-tag-version ${{ fromJson(steps.pkg.outputs.json).version }}-$(git rev-parse --short HEAD)+${{ github.run_id }}.${{ github.run_attempt }}
```

Comment:
About `--no-git-tag-version`, I don't have a strong opinion here but I know a lot of people like to be able to view the code for a version in GitHub and some users have complained about this for betas of dd-trace since betas are usually published manually and we don't add a tag. Any specific reason not to add a tag?

Reply:
If I recall correctly, the `npm version` will create a new commit and tag _that_, so pushing a tag for it would link to nothing unless we pushed the commit too, which I doubt we want to do on every commit. We aren't pushing anything here though, so I'm not sure we even _need_ the tag skipping though? 🤔

Reply:
I guess we'd need to push the commit without the branch somehow yes. Anyway, not a blocker, just a thought so that it's easier to track the code for a beta release.